File: apps/api/app/schemas/dataset.py
from datetime import datetime
from typing import List, Optional

from pydantic import BaseModel


class Dataset(BaseModel):
    id: str
    source_id: str
    name: str
    fetched_at: Optional[datetime] = None
    row_count: Optional[int] = None
    columns: Optional[List[str]] = None
    raw_path: Optional[str] = None
    cleaned_path: Optional[str] = None
    status: str = "pending"


class DatasetList(BaseModel):
    datasets: List[Dataset]
    total: int

File: apps/web/src/types/dataset.ts
export interface Dataset {
  id: string
  source_id: string
  name: string
  fetched_at?: string
  row_count?: number
  columns?: string[]
  raw_path?: string
  cleaned_path?: string
  status: string
}

export interface DatasetList {
  datasets: Dataset[]
  total: number
}

export interface DatasetPreview {
  columns: string[]
  rows: Record<string, unknown>[]
  preview_row_count: number
  total_row_count: number | null
}
